server/src/main/java/org/apache/iotdb/db/engine/storagegroup/timeindex/FileTimeIndex.java

```diff
@@ -153,15 +153,25 @@ public long getTimePartition(String tsfilePath) {
   }
 
   @Override
-  public long getTimePartitionWithCheck(String tsfilePath) throws PartitionViolationException {
+  public long getTimePartitionWithCheck(String tsFilePath) throws PartitionViolationException {
     long startPartitionId = StorageEngine.getTimePartition(startTime);
     long endPartitionId = StorageEngine.getTimePartition(endTime);
     if (startPartitionId != endPartitionId) {
-      throw new PartitionViolationException(tsfilePath);
+      throw new PartitionViolationException(tsFilePath);
     }
     return startPartitionId;
   }
 
+  @Override
+  public boolean isSpanMultiTimePartitions(String tsFilePath) {
+    try {
+      getTimePartitionWithCheck(tsFilePath);
+    } catch (PartitionViolationException e) {
+      return false;
```

Comment:
Frankly, I don't recommend returning method values in a catch block. :)
I have an idea to do the following：

```
  @Override
  public long getTimePartition(String tsFilePath) throws PartitionViolationException {
    Pair<Boolean, Long> pair = getTimePartitionWithCheck();
    if(pair.left) {
      throw new PartitionViolationException(tsFilePath);
    }else {
      return pair.right;
    }
  }

  public Pair<Boolean, Long> getTimePartitionWithCheck() {
    Pair<Boolean, Long> pair = new Pair(true, -1);
    for (int index : deviceToIndex.values()) {
      long p = StorageEngine.getTimePartition(startTimes[index]);
      if ( pair.right == -1) {
        pair.right = p;
      } else {
        if ( pair.right != p) {
          return pair;
        }
      }
      p = StorageEngine.getTimePartition(endTimes[index]);
      if ( pair.right != p) {
        return pair;
      }
    }
    if ( pair.right == -1) {
      return pair;
    }
    pair.left = false;
    return pair;
  }
```

Of course，I know that there is already an implementation of 'gettimepartition(..)'. We can take advantage of this opportunity to refactor it. What do you think？

Reply:
Hi,  I think the new `getTimePartition()` method will cost more time compare to the original one, because the new method above will iterate through all `deviceToIndex` arrays and calculate `deviceToIndex.size()` times to get the time partition id,  however in the original method, it will only calculate once. So I prefer to reserve the original method. And I have rearranged the code when checking or getting the time partition id, please take a look at it again.
Thanks~

Reply:
Yes, the  code above is just the sample code. What we most concerned about here is about removing the business logic in catch block. i.e., my suggestion is that  return `Pair<Boolean, Long> ` in `getTimePartitionWithCheck()` for remove the logic in catch block. About 'getTimePartition(..)',  both keep the original implementation or make a little improvement, such as' getTimePartition (string tsfilepath, Boolean needCheck) throws partitionviationexception `. I respect your opinion. What do you think? :)

Reply:
Yes, thanks for your careful review, I agree with your opinion, the new commit here removed the business logic in the catch block and use -1 to indict the tsFile that may cross multiple time partitions. 
As for `getTimePartition()` method ,  I prefer reserve the original two methods, one the `getTimePartition()` and the other is `getTimePartitionWithCheck() throws PartitionViolationException `, as almost method in the engine called is `getTimePartition()` method(Like the following picture shows, the method is called 37 times), the  `getTimePartitionWithCheck() throws PartitionViolationException` method only used when load one tsfile(Just 2 times ). If we combine the two method together such as `getTimePartition (string tsfilepath, Boolean needCheck) throws PartitionViolationException`, the caller may need to put the `needCheck` parameter all the time. it may be more complicated.
What do you think?
![image](https://user-images.githubusercontent.com/6237070/113371019-d6526180-9397-11eb-8981-59846eb7c10d.png)
![image](https://user-images.githubusercontent.com/6237070/113371102-0e59a480-9398-11eb-8d59-5597af8e976e.png)